src/server/routes/viewsRouter.tsx

```diff
@@ -63,89 +55,59 @@ const viewRouter = async (req: Request, res: Response) => {
       }
     } catch (error) {
       // TODO: this would be a good place to log out the error when we set up loggers
-      status = 500;
+      statusCode = 500;
     }
   }
 
   const ReactApp = (
     <Provider store={reduxStore}>
       <StaticRouter location={req.url}>
-        <HelmetProvider context={helmetContext}>
+        <Html
+          assets={getTransferableAssetsManifest(assetsManifest)}
+          serverSideReduxState={reduxStore.getState() as unknown as JSONValue}
+          serverSideConfig={configForClient}
+        >
           <Root />
-        </HelmetProvider>
+        </Html>
       </StaticRouter>
     </Provider>
   );
 
-  const jsx = extractor.collectChunks(ReactApp);
-
-  const markup = renderToString(jsx);
-
-  // Extract data after the React context has been populated during rendering
-  const { helmet } = helmetContext as FilledContext;
-  const reduxState = reduxStore.getState();
-
-  const responseString = `
-    <!DOCTYPE html>
-    <html lang="en">
-    <head>
-      <meta charset="utf-8">
-      <base href="/">
-      ${helmet.title.toString()}
-      ${helmet.meta.toString()}
-
-      ${extractor.getLinkTags()}
-      ${extractor.getStyleTags()}
-      <script>
-        window.__PRELOADED_STATE__ = ${JSON.stringify(reduxState)}
-        window.${CONFIG_FIELD_ON_WINDOW} = ${JSON.stringify(configForClient)}
-      </script>
-      <script nomodule>
-        window.location.replace("/unsupported-browser");
-      </script>
-      ${hotjarTrackingScript}
-    </head>
-    <body>  
-      <div id="ens-app" class="ens-app">${markup}</div>
-    
-      ${extractor.getScriptTags()}
-    </body>
-    </html>
-  `;
-
-  if (matchedPageConfig.path === '*') {
-    // TODO: Eventually, we will also need to return a 404 status code
-    // if the data loader explicitly tells us so
-    // (e.g. when the route is correct but a gene doesn't exist)
-    res.status(404);
-  }
+  const stream = renderToPipeableStream(ReactApp, {
+    bootstrapScripts: getBootstrapScripts(assetsManifest),
+    onShellReady() {
+      // If something errored before we started streaming, we set the error code appropriately.
+      res.statusCode = didError ? 500 : statusCode;
+      res.statusCode = statusCode;
+      res.setHeader('Content-type', 'text/html');
+      stream.pipe(res);
+    },
+    onError(x) {
+      didError = true;
+      console.error(x); // TODO: use a proper logger here
+    }
+  });
+};
 
-  res.status(status).send(responseString);
+const getBootstrapScripts = (assetsManifest: Record<string, string>) => {
+  // In development environment, the only entry point is the client.js file (it's huge and contains the runtime, third-party libs, and css)
+  // In production, webpack will code-split, and extract vendors.js and runtime-client.js chunks
+  return [
+    assetsManifest['client.js'],
+    assetsManifest['vendors.js'],
+    assetsManifest['runtime~client.js']
```

Comment:
I see the tilde used here instead of the dash. Is this intentional?

Reply:
I didn't even notice it before you mentioned it; but yes; this is the name of the key that webpack generates in its manifest, and which I copied out of the manifest file. I don't know why it replaces hyphens with tildes.

![image](https://user-images.githubusercontent.com/6834224/203276202-8d0ccfcc-5fb0-4d5a-b115-8b004c2732b4.png)

Reply:
@imransl there seems to be a problem with the screenshot (i didn't know this is even possible). The manifest file actually says this: 
```
{
  "client.css": "/static/client.8430ec0c8e0a9769a447.css",
  "client.js": "/static/client.30cdeb509016a05f4881.js",
  "unsupportedBrowser.css": "/static/unsupportedBrowser.7a874b03bf636ac9519c.css",
  "unsupportedBrowser.js": "/static/unsupportedBrowser.3423f30df83b48f77fd6.js",
  "runtime~client.js": "/static/runtime~client.51c005dbd8799cc97d18.js",
  "runtime~unsupportedBrowser.js": "/static/runtime~unsupportedBrowser.f3b0f2b750aee9b85d82.js",
  "771.css": "/static/771.9f148a96bff8837bd91e.css",
  "771.js": "/static/771.948ae1eccfce94966307.js"
  ...
}
```

Reply:
Also; including `runtime~client.js` is a requirement for the app to bootstrap (tested this in a build that did not include this file). This file is not imported independently from any other file.

Reply:
Wow that is some nice graphic processing by MacOS to change the characters like that in case you didn't accidentally change it 😄 

On a different note, that's a very interesting file naming convention.